core/roboconf-agent/src/main/java/net/roboconf/agent/internal/misc/AgentUtils.java

```diff
@@ -112,6 +118,39 @@ public static void copyInstanceResources( Instance instance, Map<String,byte[]>
 
 
 	/**
+	 * Executes a script resource on a given instance.
+	 * @param instance an instance
+	 * @throws InterruptedException
+	 * @throws IOException
+	 */
+	public static List<ExecutionResult> executeScriptResources( Instance instance, String applicationName, String scopedInstanceName ) throws IOException, InterruptedException {
+
+		File dir = InstanceHelpers.findInstanceDirectoryOnAgent( instance );
+		Utils.createDirectory(dir);
+		List<File> scriptFiles = Utils.listAllFiles( dir, Constants.FILE_EXT_SCRIPT);
```

Comment:
How does it work if there are several scripts, one invoking another one?
There should be only one script to execute. The real question is how to find the right one? You must find a convention here, knowing it cannot be about the name itself. Maybe you can think about a suffix. Example: `whatever.configure.sh` or `whatever.configure.py`. This could be a way to find the right one.